TempleEight/docs/guide/authentication.md

```diff
@@ -3,3 +3,186 @@ id: authentication
+Now that we are able to obtain an access token, we can make requests to other services by including the auth token in the request headers.
+We use the Bearer Token framework, as defined in [RFC6750](https://tools.ietf.org/html/rfc6750).
+
+Let's consider a full example of registration, followed by a request to `ExampleService`:
+
+```bash
+❯❯❯ curl -X POST $KONG_ENTRY/api/auth/register -d '{"email": "test@temple.com", "password": "abcdefgh"}'
+{"AccessToken":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1ODgwMTI3OTAsImlkIjoiNGZmYjAyZDQtODdlZC0xMWVhLWE4NjktMDI0MmFjMWYwMDAzIiwiaXNzIjoiaDFNT3hwN2lhTFFOTFA4ek1RS3k2VEkzcmpuNTlsM2MifQ.xNcAdY0r98J7rzNeEWSUPDTUg5HXOCFh41ZB74tTSw0"}
+
+❯❯❯ curl -X POST $KONG_ENTRY/api/example-service -d '{"foo": "hello", "bar": 123}' -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1ODgwMTI3OTAsImlkIjoiNGZmYjAyZDQtODdlZC0xMWVhLWE4NjktMDI0MmFjMWYwMDAzIiwiaXNzIjoiaDFNT3hwN2lhTFFOTFA4ek1RS3k2VEkzcmpuNTlsM2MifQ.xNcAdY0r98J7rzNeEWSUPDTUg5HXOCFh41ZB74tTSw0"
+{"id":"74045036-87ed-11ea-9edc-0242c0a81003","foo":"hello","bar":123}
+
+```
+Now we included a valid token in the Authorization header, the request was authorized and the entity successfully created.
+
+## Services with `#auth` Metadata
+You may have noticed that so far we have ignored the `ExampleUser` service we previously defined. 
+That's because it acts as any other service, so any example given in this guide could have been replace with a call to that service too.
+
+However, it has one distinguishing feature: a single token is only able to create a single entity in this service. 
+This means means that there is a one-to-one mapping between the `Auth` service, and an entity in the `ExampleUser` service.
+This makes the `ExampleUser` service perfect for storing additional metadata about a user, such as their name, address or anything that fits your business needs.
+
+## How Authentication is Implemented
+Behind the scenes, lots of additional code is executed when authentication is added to your project, to protect your data and your users.
+To best explain this let's consider how a new user registers with your service:
+
+#### Pre-request Preparation
+Before any request is even issued to your `Auth` service, several steps are taken:
+- The `Auth` service registers as a consumer with Kong ([see the Kong documentation](https://docs.konghq.com/2.0.x/getting-started/adding-consumers/#1-create-a-consumer-through-the-restful-api))
+- The `Auth` service requests a new HS256 JWT credential from Kong, used to sign each token
+- All other services, other than `Auth`, are configured through Kong to require a JWT as part of the request body
+
+
+#### Making a Register Request
+Upon making a request to the `/register` endpoint, the service will:
+- Validate the email and password
+  - We currently impose a minimum 8 character limit on the passwords, but you can extend these in your custom defined hooks - see the [Hooks](hooks) guide for more.
```

Comment:
How would you go about reducing this limit in a hook?

Reply:
You can't reduce it, only strengthen it